### ISSUE TITLE
Flatten CalculusGrapherControlPanelOptions. curvePushButtonGroupOptions to smoothButtonVisible

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/calculus-grapher#118]
I think we talked about this briefly in our pairing sessions last week...

In CalculusGrapherControlPanel.ts, this usage of nested options feels like it can be flattened to a single option:

```diff
-      curvePushButtonGroupOptions: {
-        smoothButtonOptions: {
-          baseColor: PhetColorScheme.BUTTON_YELLOW,
-          visible: false
-        }
-      },
+     smoothButtonVisible: false,
```

... with `baseColor: PhetColorScheme.BUTTON_YELLOW` relocated elsewhere.

**Comment by pixelzoom** [comment]
11/29/22 meeting: @veillette is going to handle this, and look for other places where nested options might be overkill.

**Comment by pixelzoom** [comment]
Here's another example in CalculusGrapherScreenView.ts.  It's repeated 4x, with a different stroke value. I don't know if the nesting is necessary or useful.

```
        curveNodeOptions: {
          continuousLinePlotOptions: {
            stroke: CalculusGrapherColors.originalCurveStrokeProperty
          }
        },
```

**Comment by veillette** [comment]
I addressed the two previous cases. I created a ` smoothButtonVisible` field. I did not flatten the structure all the way though because I also changed the default value of smoothButtonVisible, by assuming the 'Advanced` and `Lab` screens are using the  "default" configuration of the panel.  This allows me to group the fields for the `Derivative` and `Integral` screens.

In this way,  the options in `DerivatriveScreenView.js`, look like 

```js
      controlPanelOptions: {
        tangentCheckboxProperty: new BooleanProperty( true ),
        smoothButtonVisible: false
      }
```
The base color of the smooth button was moved downstream to CurvePushButtonGroup.

For the curve stroke I opted for  `continuousLinePlotStroke`. It lives in `GraphNode.js` . While I was working on it, I made `continuousLinePlotStroke`  a required option as, in this simulation,  the curve color is a very intrinsic property of a given curve.   

Assigning to @pixelzoom for review.
